sklearn/feature_selection/_multivariate_selection.py

```diff
@@ -0,0 +1,214 @@
+    >>> X, y = load_digits(return_X_y=True)
+    >>> X.shape
+    (1797, 64)
+    >>> X_new = MultivariateFeatureSelector(k = 7).fit_transform(X, y)
+    >>> X_new.shape
+    (1797, 7)
+    """
+    
+    def __init__(self, k=10):
+        self.k = k
+        
+    class _Parallel:
+        #helper class for calculating, in parallel, 
+        #test statistic associated with
+        #selected best features and an additional feature 
+        def __init__(self, X_new, y,best_features):
+            self.X_new = X_new
+            self.y = y
+            self.best_feat = best_features
+
+        def __call__(self, index):
+            if np.var(self.X_new[:,index]) == 0:
+                stat = -1000.0
+            else:   
+                if len(self.best_feat)==0:
+                    X_j =  self.X_new[:,index] 
+                    stat= k_sample_test(X_j,self.y)
+                else:
+                    columns = self.best_feat.copy() 
+                    columns.append(index)
+                    X_j = self.X_new[:,columns]
+                    stat= k_sample_test(X_j,self.y)
+            return stat
+        
+        
+    def fit(self, X, y,workers = -1):
+        """Learn the features to select from X.
+        
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of predictors.
+        y : array-like of shape (n_samples,), default=None
+            Target values. 
+        
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        if isspmatrix(X) == True:
+            X = X.toarray()
+        
+        features = np.arange(X.shape[1])
+        
+        if np.isnan(X).any() == True:
+            raise ValueError("existing non-parametric multivariate independence tests in scipy do not allow nan")
+        if type(self.k) != int:
+            raise ValueError("k features to select must be integer")
+        if not 0 <= self.k <= X.shape[1]:
+                raise ValueError("k features to select must be a non-negative integer that is less than or equal to n_features of X")
+        if not X.shape[0] >= 5:
+                raise ValueError("n_samples of data matrix X must be >= 5 in order to perform k_sample multivariate independence test")
+        
+        #loop to select k best features, 
+        #each iteration adds next best feature
+        best_features = []
+        while len(best_features) < self.k: 
+            X_new = np.array(X)
+            
+            #Mapwrapper parallelizes test statistic calculations 
+            #of selected best features and each additional feature.
+            #size of operations in parallel per loop iteration is 
+            #n_features - len(best_features)
+            parallel = self._Parallel(X_new=X_new, y=y,best_features = best_features)
```

Comment:
Since scikit-learn has joblib as a dependency, which in my opinion is better than multiprocessing. Do something like [this](https://github.com/neurodata/hyppo/blob/135cfcd6b13a986b09ec7629931bce8676ea5547/hyppo/tools/common.py#L456) which calls [this](https://github.com/neurodata/hyppo/blob/135cfcd6b13a986b09ec7629931bce8676ea5547/hyppo/tools/common.py#L368) for a single permutation

Reply:
space after `,` and remove spaces around `=`

Reply:
https://github.com/srecupe2/scikit-learn/commit/0cca1d0f8c89244981a30c22928271880153b171